astra-sim/system/PacketBundle.cc

```diff
@@ -54,10 +54,15 @@ void PacketBundle::send_to_NPU() {
 void PacketBundle::call(EventType event, CallData* data) {
   if (needs_processing == true) {
     needs_processing = false;
+    if (sys->mem == nullptr){
```

Comment:
From my understanding, ASTRA-sim always has a memory model. Memory latency is used to calculate latencies for collective communication. Could you please confirm this when you get a chance, @rashidi1saeed?